pkg/config/blockfetcher_config.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+const (
+	DefaultTimeout                = 1 * time.Minute
+	DefaultOIDBatchSize           = 8000
+	DefaultDownloaderWorkersCount = 500
+)
+
+// NeoFSBlockFetcher represents the configuration for the NeoFS block fetcher service.
+type NeoFSBlockFetcher struct {
+	InternalService        `yaml:",inline"`
+	Timeout                time.Duration `yaml:"Timeout"`
+	ContainerID            string        `yaml:"ContainerID"`
+	Addresses              []string      `yaml:"Addresses"`
+	OIDBatchSize           int           `yaml:"OIDBatchSize"` // valid only for SkipIndexFilesSearch = true
+	BlockAttribute         string        `yaml:"BlockAttribute"`
+	OidAttribute           string        `yaml:"OidAttribute"`
+	HeaderAttribute        string        `yaml:"HeaderAttribute"`
+	DownloaderWorkersCount int           `yaml:"DownloaderWorkersCount"`
+	BQueueSize             int           `yaml:"BQueueSize"`
+	SkipIndexFilesSearch   bool          `yaml:"SkipIndexFilesSearch"`
+}
+
+// Validate checks NeoFSBlockFetcher for internal consistency and ensures
+// that all required fields are properly set. It returns an error if the
+// configuration is invalid or if the ContainerID cannot be properly decoded.
+func (cfg *NeoFSBlockFetcher) Validate() error {
+	if !cfg.Enabled {
+		return nil
+	}
+	if cfg.ContainerID == "" {
+		return errors.New("container ID is not set")
+	}
+	var containerID cid.ID
+	err := containerID.DecodeString(cfg.ContainerID)
+	if err != nil {
+		return fmt.Errorf("invalid container ID: %w", err)
+	}
+	if cfg.Timeout == 0 {
+		cfg.Timeout = DefaultTimeout
+	}
+	if cfg.OIDBatchSize == 0 {
+		cfg.OIDBatchSize = DefaultOIDBatchSize
+	}
+	if cfg.DownloaderWorkersCount <= 0 {
```

Comment:
Why not to move all these overrides to `newServerFromConstructors`? It's a common way for our services to override unset values with defaults in constructors.

Reply:
E.g.:
```
	if config.ExtensiblePoolSize <= 0 {
		config.ExtensiblePoolSize = defaultExtensiblePoolSize
		log.Info("ExtensiblePoolSize is not set or wrong, using default value",
			zap.Int("ExtensiblePoolSize", config.ExtensiblePoolSize))
	}
```